cimbar/encode/cimb_translator.py

```diff
@@ -121,40 +132,64 @@ def _check_color(self, c, d):
         #return (c[0] - d[0])**2 + (c[1] - d[1])**2 + (c[2] - d[2])**2
         return relative_color_diff(c, d)
 
-    def _scale_color(self, c, adjust, down):
+    def _scale_adjust(self, c, adjust, down):
         c = int((c - down) * adjust)
-        if c > (245 - down):
+        if c > 245:
             c = 255
+        if c < 0:
+            c = 0
         return c
 
-    def _correct_all_colors(self, r, g, b):
-        if self.ccm is not None:
-            r, g, b = self.ccm.dot(numpy.array([r, g, b]))
-        return r, g, b
-
-    def _best_color(self, r, g, b):
-        r, g, b = self._correct_all_colors(r, g, b)
-
-        # probably some scaling will be good.
+    def scale_color(self, r, g, b):
+        if self.disable_color_scaling:
+            return r, g, b
         if self.dark:
             max_val = max(r, g, b, 1)
-            min_val = min(r, g, b, 48)
+            min_val = min(r, g, b, max_val-100)
             if min_val >= max_val:
                 min_val = 0
             adjust = 255.0 / (max_val - min_val)
-            r = self._scale_color(r, adjust, min_val)
-            g = self._scale_color(g, adjust, min_val)
-            b = self._scale_color(b, adjust, min_val)
+            r = self._scale_adjust(r, adjust, min_val)
+            g = self._scale_adjust(g, adjust, min_val)
+            b = self._scale_adjust(b, adjust, min_val)
         else:
             min_val = min(r, g, b)
             max_val = max(r, g, b, 1)
             if max_val - min_val < 20:
                 r = g = b = 0
             else:
                 adjust = 255.0 / (max_val - min_val)
-                r = self._scale_color(r, adjust, min_val)
-                g = self._scale_color(g, adjust, min_val)
-                b = self._scale_color(b, adjust, min_val)
+                r = self._scale_adjust(r, adjust, min_val)
+                g = self._scale_adjust(g, adjust, min_val)
+                b = self._scale_adjust(b, adjust, min_val)
+        return r, g, b
+
+    def _correct_all_colors(self, r, g, b, sector):
+        if isinstance(self.ccm, list):
+            ccm = self.ccm[sector]
```

Comment:
Similarly, this is another "well what if we want to..." -- in this case, using the naive ccm based on a single color average (white) followed by a von kries transform, then feeding *those* corrected values into the Moore-Penrose based CCM, then smashing the resulting CCMs together for one king of all CCMs.

Anyway I don't think this one is that useful at this point, and it's about as silly as that explanation makes it sound. So this one will probably get removed.